### ISSUE TITLE
[Bug]: 0.19.0-rc.1 through 5:  Non-admin moderators cannot list reports.

### ISSUE THREAD

**Issue by asimons04** [LemmyNet/lemmy#4172]
### Requirements

- [X] Is this a bug report? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a single bug? Do not put multiple bugs in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Summary

The `/post/report/list` and `/coment/report/list` endpoints are returning a 400: "not an admin" error when regular community moderators try to view reports.

Additionally, calls to `/user/report_count` returns the same 400 error with "not an admin" as the message.

Have noticed this since at least 0.19.0-rc.1.

Have confirmed with direct API calls as well as on Voyager's Lemmy-UI.

### Steps to Reproduce

1. Create an account on voyager.lemmy.ml (that is not an admin)
2. Create a community or be added as a moderator to any community
3. Try to load the reports while inspecting the network requests in the browser dev console
4. Alternatively, make authenticated API calls to the report list/count endpoints from Postman.

### Technical Details

Tested against the latest instance at voyager.lemmy.ml using both Voyager's Lemmy-UI and via direct API call using the `Auth Bearer:` method (which successfully authenticates to other API endpoints).

### Version

BE: 0.19.0-rc.5

### Lemmy Instance URL

voyager.lemmy.ml

**Comment by asimons04** [comment]
Confirmed to exist in the wild.  Tested against lemmy.map-p.me with a local user, mod but not admin, against a test community.  Same behavior as described in bug.

Additionally, maybe related, but did not receive any email notification for the report. SMTP on that instance was configured correctly as I did receive the approval email moments prior to that test.

**Comment by maxpoulin64** [comment]
Not seeing anything weird in the logs, basically what you'd expect to see reported:

```
2023-11-17T23:22:53.831806Z  WARN lemmy_server::root_span_builder: NotAnAdmin: NotAnAdmin
   0: lemmy_api::local_user::report_count::report_count
           with data=Query(GetReportCount { community_id: None }) local_user_view=LocalUserView { local_user: LocalUser { id: LocalUserId(7), person_id: PersonId(2383864), password_encrypted: "REDACTED", email: Some("REDACTED"), show_nsfw: false, theme: "browser", default_sort_type: Active, default_listing_type: Local, interface_language: "browser", show_avatars: true, send_notifications_to_email: true, show_scores: true, show_bot_accounts: true, show_read_posts: true, email_verified: false, accepted_application: true, totp_2fa_secret: None, open_links_in_new_tab: false, blur_nsfw: true, auto_expand: false, infinite_scroll_enabled: false, admin: false, post_listing_mode: List, totp_2fa_enabled: false, enable_keyboard_navigation: false, enable_animated_images: true, collapse_bot_comments: false }, person: Person { id: PersonId(2383864), name: "admiralpatrick", display_name: None, avatar: None, banned: false, published: 2023-11-17T23:07:55.965823Z, updated: None, actor_id: DbUrl(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("lemmy.max-p.me")), port: None, path: "/u/admiralpatrick", query: None, fragment: None }), bio: None, local: true, private_key: Some("REDACTED"), public_key: "REDACTED", last_refreshed_at: 2023-11-17T23:07:55.965823Z, banner: None, deleted: false, inbox_url: DbUrl(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("lemmy.max-p.me")), port: None, path: "/u/admiralpatrick/inbox", query: None, fragment: None }), shared_inbox_url: Some(DbUrl(Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("lemmy.max-p.me")), port: None, path: "/inbox", query: None, fragment: None })), matrix_user_id: None, bot_account: false, ban_expires: None, instance_id: InstanceId(1) }, counts: PersonAggregates { person_id: PersonId(2383864), post_count: 0, post_score: 1, comment_count: 1, comment_score: 1 } }
             at crates/api/src/local_user/report_count.rs:15
   1: lemmy_server::root_span_builder::HTTP request
           with http.method=GET http.scheme="http" http.host=lemmy.max-p.me http.target=/api/v3/user/report_count otel.kind="server" request_id=1f159ec6-2453-47be-ac74-78a9a98635ea
             at src/root_span_builder.rs:16

```

**Comment by dessalines** [comment]
Was an issue with an additional mod check that got added with some incorrect logic around the community checking. I have a PR fix up.
